cache/threadsafelrucache.go

```diff
@@ -66,3 +75,15 @@ func (c *ThreadSafeLruCache) Clear() {
 	defer c.mutex.Unlock()
 	c.cache.Clear()
 }
+
+// Range all stored items from the cache.
+func (c *ThreadSafeLruCache) Range(f func(key lru.Key) bool) {
+	c.mutex.RLock()
+	keys := c.cache.Keys()
+	c.mutex.RUnlock()
```

Comment:
better to use defer c.mutex.RUnlock()

otherwise, another thread can kick in and delete one key before the for loop is executed